staging/src/k8s.io/apimachinery/pkg/util/proxy/upgradeaware_test.go

```diff
@@ -246,6 +247,20 @@ func TestServeHTTP(t *testing.T) {
 			expectedPath:    "/some/path",
 			useLocationHost: true,
 		},
+		{
+			name:               "append server path to request path",
+			method:             "GET",
+			requestPath:        "/base",
+			expectedPath:       "/base/base",
```

Comment:
Add a test where this expected path ends in a `/`, so that if someone "helpfully" replaces singleJoiningSlash with path.Join, they will break a test.